### ISSUE TITLE
XPath errors are logged but do not show the expression and location

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#3351]
Example of error:

```
|----------------------------------------------------------------------------------------------------------------------|
|[No error message provided.]                                                                                          |
|----------------------------------------------------------------------------------------------------------------------|
|Application Call Stack                                                                                                |
|----------------------------------------------------------------------------------------------------------------------|
|----------------------------------------------------------------------------------------------------------------------|
|Exception: org.orbeon.saxon.dom4j.TypedNodeWrapper$TypedValueException                                                |
|----------------------------------------------------------------------------------------------------------------------|
|org.orbeon.saxon.dom4j.TypedNodeWrapper            |atomize                       |TypedNodeWrapper.java         | 133|
|org.orbeon.saxon.expr.CastableExpression           |effectiveBooleanValue         |CastableExpression.java       | 179|
```

[+1 from customer](https://mail.google.com/mail/u/0/#inbox/15f0d322956950da)

**Comment by avernet** [comment]
[+1 from customer](https://3.basecamp.com/3600924/buckets/2825555/messages/1841506774)

**Comment by avernet** [comment]
[+1 from customer](https://3.basecamp.com/3600924/buckets/7873395/messages/1928962952)

**Comment by avernet** [comment]
[+1 from user](http://discuss.orbeon.com/feature-request-point-to-xpath-expression-that-is-incorrect-td4665539.html)

**Comment by avernet** [comment]
The last change doesn't handle all cases, but should already be helpful.

**Comment by avernet** [comment]
Closing days, as most cases should be handled, and we can always create new issues in case we find other situations that are not handled by this.

**Comment by avernet** [comment]
I also cherry-picked this on the `2020.1-pe` branch, to be included in 2020.1.4.
